### ISSUE TITLE
Please contribute: Instance spawn data mapping

### ISSUE THREAD

**Issue by Treeston** [TrinityCore/TrinityCore#20108]
_(84590be to shut the bot up)_

This issue exists to track progress on gathering instance spawn association - e.g. "which trash spawns belong to which boss" - data.

As outlined [here](https://community.trinitycore.org/topic/13650-the-great-instance-spawn-crowdsource-initiative-of-2017/), this is a very accessible process that anyone can contribute to. Please follow the steps outlined in the thread linked and post your results either in that thread or directly as a comment on this issue.

Data will be merged into the 3.3.5 branch as soon as we have a given instance mapped by three different sources (to avoid introducing hard-to-find issues). A full overview of instance progress can be found below:

<table>
<tr><th>Instance</th><th>1</th><th>2</th><th>3</th></tr>
<tr><th colspan=4>WotLK Raids</th></tr>
<tr><td>Ruby Sanctum</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Icecrown Citadel</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Onyxia's Lair</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Ulduar</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Obsidian Sanctum</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Eye of Eternity</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Naxxramas</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>Vault of Archavon</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><th colspan=4>WotLK Dungeons</th></tr>
<tr><td>Utgarde Keep</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>The Nexus</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Azjol'Nerub</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>Ahn'kahet</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Drak'tharon Keep</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Gundrak</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Halls of Stone</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Halls of Lightning</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Oculus</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Utgarde Pinnacle</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Forge of Souls</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Pit of Saron</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Halls of Reflection</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><th colspan=4>BC Raids</th></tr>
<tr><td>Sunwell Plateau</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Black Temple</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Tempest Keep: The Eye</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>Serpentshrine Cavern</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Zul'Aman</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Gruul's Lair</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>Magtheridon's Lair</td><td>✔</td><td>✘</td><td>✘</td></tr>
<tr><td>Karazhan</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><th colspan=4>BC Dungeons</th></tr>
<tr><td>Hellfire Ramparts</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Blood Furnace</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Slave Pens</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Underbog</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Mana-Tombs</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Auchenai Crypts</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Caverns of Time: Old Hillsbrad</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Sethekk Halls</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Shadow Labyrinth</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Steamvault</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Shattered Halls</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Tempest Keep: The Arcatraz</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Tempest Keep: The Mechanar</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Tempest Keep: The Botanica</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><th colspan=4>Vanilla Raids</th></tr>
<tr><td>Temple of Ahn'Qiraj</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Ruins of Ahn'Qiraj</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Blackwing Lair</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Molten Core</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Zul'Gurub</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><th colspan=4>Vanilla Dungeons</th></tr>
<tr><td>Blackrock Spire</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Blackrock Depths</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Dire Maul</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Scholomance</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Stratholme</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Zul'Farrak</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Uldaman</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Razorfen Downs</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Sunken Temple</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Blackfathom Deeps</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Razorfen Kraul</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Stockades</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Scarlet Monastery</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Gnomeregan</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Shadowfang Keep</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Wailing Caverns</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>The Deadmines</td><td>✘</td><td>✘</td><td>✘</td></tr>
<tr><td>Ragefire Chasm</td><td>✘</td><td>✘</td><td>✘</td></tr>
</table>

**Comment by bumbummen99** [comment]
I will take a look into it once my vacation has started

**Comment by necropola** [comment]
I would like to help and have set it all up (current TC-3.3.5 + Addon), but have a few questions
- Addon: Is there a way to delete Spawn Groups made by accident?
- Addon: What is the purpose of the context menu to the right of a group? I guess it's to assign all members of that group to a different group. It seems it can also have the effect of deleting a group and/or messing things up entirely,
![image](https://user-images.githubusercontent.com/7461137/29004440-35f26272-7ac7-11e7-888f-edf403bb79a7.png)
- Addon: What do the different colors mean?
- Addon: Is there a way (command) to reset all data for a given instance and start over from scratch?
- General: Isn't it a bit random which NPCs I assign to which group in an instance like Ragefire Chasm? Can you elaborate a bit more on how you want them assigned? Like everything up to first boss assign to first boss, and so on, ...

**Comment by Treeston** [comment]
> Addon: Is there a way to delete Spawn Groups made by accident?

Move the entire group to another group (or to Unassigned). This will delete the group (even if there's nothing in it).

> Addon: What is the purpose of the context menu to the right of a group? I guess it's to assign all members of that group to a different group. It seems it can also have the effect of deleting a group and/or messing things up entirely,

Assigns everything in that group to a different group, then deletes the group. If it "messes things up entirely", that's a bug, please tell me about it.

> What do the different colors mean?

They're random, leftover from when I shaded the entries' backgrounds - I ended up not doing it because `ScrollFrame` has some _really_ weird texture rendering glitches.

> Is there a way (command) to reset all data for a given instance and start over from scratch?

While you have the frame open, you can run this to reset current map data:
```lua
/run SpawnGroupMapper.savedData = {lookup={},groups={}} ReloadUI()
```

> Isn't it a bit random which NPCs I assign to which group in an instance like Ragefire Chasm? Can you elaborate a bit more on how you want them assigned? Like everything up to first boss assign to first boss, and so on, ...

Instances like RFC are a bit random in that you can't really tell, and we have no way to check on retail (RFC was redesigned). Generally, all trash packs should be assigned to the subsequent boss...unless they have special behavior for some reason.

For RFC, the obvious ones are the Troggs (Trogg boss) and the side path (Satyr boss). Everything else is a judgment call since Felguard boss is on his central platform while Warlock boss shares his trash - I think I assigned anything "behind" Felguard to warlock, everything in front to Felguard. It's a "best guess" thing - later instances are a lot clearer.

**Comment by necropola** [comment]
I can not even uncollapse Unassiged anymore.
The command to reset the per instance data did not show any effect.
I've deleted the addon data manually.
Still no joy. Giving up ...

**Comment by Treeston** [comment]
You shouldn't even be able to _collapse_ that entry to begin with. 😮

**Comment by necropola** [comment]
I would love to get it to work, but at the moment the Addon is unusable for me. Maybe I'm too stupid or some other Addon interferes. Also, I don't see a way, to just add a currently targeted NPC to a specific group. Teleporting to an NPC from the list is nice, but then I need to figure out where I am. More than once I fell through the ground when moving after teleport ...
Don't get me wrong. I want to help getting the data, but it's very frustrating at the moment.

**Comment by qyh214** [comment]
Although I really want to be able to help, but the addon loaded, but when I `/spawngroup`,it shows  `Type /help to get the command list`(The general content of system yellow characters)

**Comment by HannibalRoG** [comment]
@Treeston i can not post here https://community.trinitycore.org/topic/13650-the-great-instance-spawn-crowdsource-initiative-of-2017/

I try to use addon with a normal account a did not work, logued in with my gm account and work fine.
This is the sql from stormwind stokades, review and tell me if it ok, then i will continue to other instances, no promises, but I think before the new year we can have them all scanned.

[StormwindStokades.txt](https://github.com/TrinityCore/TrinityCore/files/1328129/StormwindStokades.txt)

**Comment by Treeston** [comment]
@HannibalRoG looking at your sql, your groups are wrong. We're not looking to make a group per type of mob - we want one group per boss. In the Stockades case, that means we want to end up with six groups:

* Bazil Thredd
* Dextren Ward
* Hamhock
* Kam Deepfury
* Targorr the Dread
* Bruegal Ironknuckle

Each of these groups should contain the mobs that will stop respawning when the associated boss is killed.

**Comment by HannibalRoG** [comment]
Basically all the mobs that i have to kill to access the boss must go in the same group with the boss, right?
Now I have a doubt, I understand that certain mobs respawn after a while patrolling the dungeon, if that is true, how would be the respawn of such mobs?

**Comment by Treeston** [comment]
Linked to final boss, typically.

**Comment by HannibalRoG** [comment]
And how do I know which ones in a dungeon should be associated with the final boss?

**Comment by Keader** [comment]
You can use retail to do it.
Try dont kill any trash, and kill the boss. Go out of instance, wait 15~30 min (soft reset) and go inside dungeon again.
You will see alot of trashs is not there anymore :)

**Comment by Keader** [comment]
Nope, we still can access all dungeons/instance of wotlk on retail...
Blizzard dont care about change stuffs in old raids...

**Comment by HannibalRoG** [comment]
Ragefire Chasm
Shadowfang Keep
Scarlet Monastery
That three for example change all trash and most bosses, the history arround it change to, that it what i remember, i know there is much more, beside; the time, money and internet conection for expend in that job its not available for me now, i can only work with Trinitycore server to do it.
In any case, we have a group of people here very knowledgeable of all the LK content, I think we can do a good job without having access to retail.

**Comment by HannibalRoG** [comment]
Pro tip:?
Its very likely that the trash that have waypoint (patrol the instance) inside dungeon is linked to final boss.

Will this be correct?

**Comment by HannibalRoG** [comment]
First two

[RagefireChasm.txt](https://github.com/TrinityCore/TrinityCore/files/1346834/RagefireChasm.txt)
[Tempest Keep-The Eye.txt](https://github.com/TrinityCore/TrinityCore/files/1346838/Tempest.Keep-The.Eye.txt)

**Comment by qyh214** [comment]
Ok~I use en_us client and do well.
here's my first work for icc,
(I don't know if my understanding is correct)

[Icecrown Citadel.zip](https://github.com/TrinityCore/TrinityCore/files/1355216/Icecrown.Citadel.zip)

**Comment by HannibalRoG** [comment]
[Bancal del Magister.zip](https://github.com/TrinityCore/TrinityCore/files/1429677/Bancal.del.Magister.zip)
[Uploading Bancal del Magister.zip…]()
[Fortaleza de Utgarde.zip](https://github.com/TrinityCore/TrinityCore/files/1429678/Fortaleza.de.Utgarde.zip)

**Comment by HannibalRoG** [comment]
[Azjol'Nerub.zip](https://github.com/TrinityCore/TrinityCore/files/1442892/Azjol.Nerub.zip)
[Ahn'kahet.zip](https://github.com/TrinityCore/TrinityCore/files/1442895/Ahn.kahet.zip)

[Dead Mines.zip](https://github.com/TrinityCore/TrinityCore/files/1442904/Dead.Mines.zip)

**Comment by HannibalRoG** [comment]
[Gundrak.zip](https://github.com/TrinityCore/TrinityCore/files/1443011/Gundrak.zip)
[Halls of Lightning.zip](https://github.com/TrinityCore/TrinityCore/files/1443012/Halls.of.Lightning.zip)
[Halls of Stone.zip](https://github.com/TrinityCore/TrinityCore/files/1443013/Halls.of.Stone.zip)
This one is from another source [Ragefire Chams.zip](https://github.com/TrinityCore/TrinityCore/files/1443014/Ragefire.Chams.zip)

**Comment by HannibalRoG** [comment]
[Shadowfang Keep.zip](https://github.com/TrinityCore/TrinityCore/files/1443016/Shadowfang.Keep.zip)
[The Nexus-5 Heroic (Alliance).zip](https://github.com/TrinityCore/TrinityCore/files/1443017/The.Nexus-5.Heroic.Alliance.zip)
[The Nexus-5 Heroic (Horde).zip](https://github.com/TrinityCore/TrinityCore/files/1443018/The.Nexus-5.Heroic.Horde.zip)

**Comment by HannibalRoG** [comment]
[The Oculus.zip](https://github.com/TrinityCore/TrinityCore/files/1443019/The.Oculus.zip)

**Comment by HannibalRoG** [comment]
[Drak'tharon Keep.zip](https://github.com/TrinityCore/TrinityCore/files/1445281/Drak.tharon.Keep.zip)
[Pit of Saron.zip](https://github.com/TrinityCore/TrinityCore/files/1445282/Pit.of.Saron.zip)
[The Forge of Souls.zip](https://github.com/TrinityCore/TrinityCore/files/1445283/The.Forge.of.Souls.zip)

**Comment by Treeston** [comment]
Awesome, thanks so far. Just dropping by to let you know that I haven't abandoned this, and your efforts are not in vain 😉

**Comment by HannibalRoG** [comment]
@Treeston thanks, it's good to hear it, when you can, update the table.

**Comment by HannibalRoG** [comment]
WTF i cant upload any file

**Comment by Aokromes** [comment]
maybe there is a limit on atachments.

**Comment by Jildor** [comment]
Better use gist @HannibalRoG

**Comment by HannibalRoG** [comment]
[Auchenai Crypts By Psyhopath](https://gist.github.com/HannibalRoG/d590ead567f38c0b25260a02ebe2c75c)
Blackfathon deeps By Black https://gist.github.com/HannibalRoG/0a05263d0821f3dc15f4114a0147a444
Halls of Reflection By Leroux https://gist.github.com/HannibalRoG/226038619200d8442dab5e155b46a160
Hellfire Citadel By Psyhopath https://gist.github.com/HannibalRoG/288b46bb106138f119f3e6779a5b8bc1
Icecrown Citadel By Leroux https://gist.github.com/HannibalRoG/5c5efb3a2bee1ccb77ecb86e9e5d66a9
Mana-Tombs By Psyhopath https://gist.github.com/HannibalRoG/8f32c11c7a84c1db3f04c8d68067c96e
Naxxramas By Leroux https://gist.github.com/HannibalRoG/e993786da5f3bc590dcdf02b62901f42
Pit of Saron By Leroux https://gist.github.com/HannibalRoG/d164a02d3981b6df42818b0c99503450
Serpentshrine Cavern By Psyhopath https://gist.github.com/HannibalRoG/ee078e2dd39e2ed9787749b6e4c8fe4b
The Stockade By Black https://gist.github.com/HannibalRoG/b335f35159aaf50d3a97e15956f1ec5f
Shadow Labyrinth By Psyhopath https://gist.github.com/HannibalRoG/9552d59c5ade135a25520e0dd4baa8b3

**Comment by HannibalRoG** [comment]
[Vault of Archavon By Leroux](https://gist.github.com/HannibalRoG/4128ac460d1cedbe8e02bc667000626f)
 [The Forge of Souls By Leroux](https://gist.github.com/HannibalRoG/07be738a0bb240f2d782386711fe4f50)
 [The Slave By Psyhopath](https://gist.github.com/HannibalRoG/fa8398a4f0463c37c3e4037226483798)
 [The Steamvault By Psyhopath](https://gist.github.com/HannibalRoG
/1b591195f5febc746a81b1315d04a32a)
 [The Underbog By Psyhopath](https://gist.github.com/HannibalRoG/14ed43405915c4a1291789c3c96800c9)
 [Ulduar By Leroux](https://gist.github.com/HannibalRoG/a0523e0f0cfb385962de9addd37cf689)

Please note: Different name in archive=Different source.

**Comment by MantisLord** [comment]
Does anyone have a source that trash in dungeons should be associated with bosses in this manner?

For raids I know the "no respawn if boss dead" behavior is correct, but I can't find any proof on retail or online about dungeons.

**Comment by Jildor** [comment]
@Treeston I hope you will not end up losing the files, when will they apply?

**Comment by HannibalRoG** [comment]
@Treeston are you still stay with us?

**Comment by Treeston** [comment]
Yes. Most of my focus is on #19930 right now.

**Comment by chaodhib** [comment]
Shouldn't we tag this kind of issue with Github's default label's "help wanted'?

**Comment by tkrokli** [comment]
Sorry for not replying before, didn't know quite what to say.
Re-creating the labels is OK.
- ( `#159818` = [`help wanted`] )
- ( `#7962ea` or `#5319e7` = [`good first issue`] )

**Comment by chaodhib** [comment]
Thanks for replying!

**Comment by Aokromes** [comment]
I seriously don't get the point of Help-Wanted and good first issue, labels...... if you have a tracker where you accept fixes and pull requests... it's to get help.

**Comment by Treeston** [comment]
`Help-Wanted` is for issues that need unrelated users to actively contribute, I guess?

General issues are there so we can keep track of what's broken and find stuff to work on.

**Comment by chaodhib** [comment]
"good first issue" are a good way to help beginners find tasks that they can work on. It makes it easier to contribute for the first times to the project. As a beginner, it's hard to identify what's easy to fix from what is not. Only people who are knowledgeable about the system can do it.

As for "help wanted", imo it's particularly useful for PRs imo. Where the amount of work is to too much to handle for one person. In some PRs, it's possible to split the work among several people.

**Comment by chaodhib** [comment]
Someone renamed the label "help wanted" to "Help-Wanted" for a consistency reason with the other labels I assume. The issue is when going on these pages: https://github.com/TrinityCore/TrinityCore/issues or https://github.com/TrinityCore/TrinityCore/pulls, it is possible to directly filter by the label "help wanted". If the label is renamed, the whole initiative will loose visibility.

**Comment by tkrokli** [comment]
OK, so if the label properties gets messed up by renaming it (maybe losing connection with the overall github functionality, IDK), maybe it will be better to delete the misnamed one and recreate it?

**Comment by chaodhib** [comment]
No renaming the label back is fine.

**Comment by tkrokli** [comment]
OK, renaming done. Hope it helps. ;-) (pun intended)

**Comment by Heaven31415** [comment]
[sunwellplateau.zip](https://github.com/TrinityCore/TrinityCore/files/1649123/sunwellplateau.zip)

**Comment by Treeston** [comment]
List in issue updated with contributions so far (thanks!)

**Comment by DanVS** [comment]
Could you please post the tool somewhere in Github (or such, without sign up needed)?

**Comment by Treeston** [comment]
Didn't realize our forum needs you to register to see files.
[SpawnGroupMapper.zip](https://github.com/TrinityCore/TrinityCore/files/2410199/SpawnGroupMapper.zip)

**Comment by DanVS** [comment]
Thanks!

[Utgarde Keep](https://github.com/TrinityCore/TrinityCore/files/2410893/utgardekeep.txt)

**Comment by Elarose** [comment]
Great! Will jump onboard with this. =) Before starting on the older instances, has anyone a clear idea of where Trinitycore was at with creature trash pools. (Trash mobs in old instances were not static spawns, rather, a chance to spawn different creature id's within a pool. Zul'furrak even on retail still)

**Comment by offl** [comment]
https://github.com/TrinityCore/TrinityCore/issues/20108#issuecomment-347625800
I second that.

**Comment by offl** [comment]
Aokromes just tested HM HoL and trash respawned after 2 hours after killing bosses
So no linked respawn for non-raids

**Comment by Aokromes** [comment]
1 boss, not every boss / trash

**Comment by offl** [comment]
Oh, why not all trash and bosses :D
